### PR TITLE
Remove heapster support for kubectl

### LINKED ISSUE #806: Drop heapster support in kubectl top
Kubectl supports Metrics API provided by metrics server starting from 1.8
Heapster is RETIRED and no longer supported.

Having kubectl still mention heapster leads to confusion from users on what is needed to have kubectl top working.

/assign

### PR DESCRIPTION
**Pull request by serathius**
Heapster is already retired

/kind cleanup

**What this PR does / why we need it**:
Having kubectl top mention Heapster is confusing for users

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubectl/issues/806

**Special notes for your reviewer**:

```release-note
Deprecate kubectl top flags related to heapster
Drop support of heapster in kubectl top
```
/cc @brancz

### CODE REVIEW

**Comment by brancz** [comment]
I think heapster’s deprecation time is crucial here. What kube release timeframe did we deprecate heapster? As it has to be at least 4 releases ago I believe. I believe it was much longer ago but I just want to double check.

**Comment by brancz** [comment]
I just double checked, it was deprecated in 1.11 and removal was actually already planned for 1.13, so we’re good here.

https://github.com/kubernetes-retired/heapster/blob/master/docs/deprecation.md

/lgtm
/approve

File: staging/src/k8s.io/kubectl/pkg/cmd/top/top_node.go [hunk 1]
```diff
@@ -43,42 +42,13 @@ type TopNodeOptions struct {
...
-		o.Service = metricsutil.DefaultHeapsterService
-	}
-	if len(o.Scheme) == 0 {
-		o.Scheme = metricsutil.DefaultHeapsterScheme
-	}
-	if len(o.Port) == 0 {
-		o.Port = metricsutil.DefaultHeapsterPort
-	}
-
-	flags.StringVar(&o.Namespace, "heapster-namespace", o.Namespace, "Namespace Heapster service is located in")
```
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/top/top_node.go]: I do understand your approach and even though these flags might not be used, we are still obliged to follow the [deprecation policies](https://kubernetes.io/docs/reference/using-api/deprecation-policy/). Please then deprecate these flags and we'll be able to remove then a year from now.

**Review by soltysh — CHANGES_REQUESTED** [review]

File: staging/src/k8s.io/kubectl/pkg/cmd/top/top_node.go (hunk 1 above)
Comment (serathius) [reply staging/src/k8s.io/kubectl/pkg/cmd/top/top_node.go]: Ok, would it make sense to discuss exception in this case? Those flags are used exclusively component that was deprecated 6 and retired 4 releases ago. We should try to answer question "Do we have a chance to break users?". In my opinion no.

I'm for removing those flags as I had multiple users being confused by kubectl still mentioning Heapster. All kubernetes documentation already discourages usage of Heapster.

Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/cmd/top/top_node.go]: Sorry, no exception, we need to be clear, even if that means holding these flags even when they are not doing anything. Users need the time to remove them.

Comment (serathius) [reply staging/src/k8s.io/kubectl/pkg/cmd/top/top_node.go]: Ok, will prepare separate PR focused on deprecation.

Comment (serathius) [reply staging/src/k8s.io/kubectl/pkg/cmd/top/top_node.go]: Ok, I updated this PR to leave heapster flags as no-op.

**Comment by serathius** [comment]
@soltysh ping

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/priority backlog

**Comment by soltysh** [comment]
> @soltysh ping

Thanks @serathius and sorry for the delay!

**Pull request merged**